File: test90_out/local.mcu/Pin0/Pin0.js

/*
 * ======== MODULE local.mcu.Pin0 ========
 */

var $$c = function() {
    const Pin0 = {};
    Pin0.Pin0 = Pin0
    Pin0.$name = 'local.mcu.Pin0';
    Pin0.pollen$used = 0;
    Pin0.targetInit = new $$Ref('local_mcu_Pin0_targetInit__I');
    Pin0.set = new $$Ref('local_mcu_Pin0_set__E');
    Pin0.clear = new $$Ref('local_mcu_Pin0_clear__E');
    Pin0.toggle = new $$Ref('local_mcu_Pin0_toggle__E');
    Pin0.get = new $$Ref('local_mcu_Pin0_get__E');
    Pin0.makeInput = new $$Ref('local_mcu_Pin0_makeInput__E');
    Pin0.isInput = new $$Ref('local_mcu_Pin0_isInput__E');
    Pin0.makeOutput = new $$Ref('local_mcu_Pin0_makeOutput__E');
    Pin0.isOutput = new $$Ref('local_mcu_Pin0_isOutput__E');
    Pin0.pname = "Pin";
    Pin0.pnum = 0;
    Pin0.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    Pin0.pollen__uses__ = function() {
    }
    Pin0.$$privateInit = function() {
        Pin0.name = null;
        Pin0.num = 0;
        Pin0.state = 0;
        Pin0.input = 0;
        Pin0.pollen__unitname = "Pin0";
    }
    return Pin0;
}
var $$u = $$c();
$units['local.mcu.Pin0'] = $$u;
$units.push($$u);

File: test90_out/local.mcu/Pin0/Pin0.c

/*
 * ======== string literals ========
 */

static const char local_mcu_Pin0__s__fileName[] = "Pin0.p";
static const char local_mcu_Pin0__s__2[] = ".set\n";
static const char local_mcu_Pin0__s__9[] = ".isOutput\n";
static const char local_mcu_Pin0__s__6[] = ".makeInput\n";
static const char local_mcu_Pin0__s__3[] = ".clear\n";
static const char local_mcu_Pin0__s__10[] = "Pin0";
static const char local_mcu_Pin0__s__5[] = ".get\n";
static const char local_mcu_Pin0__s__0[] = "Pin Init: ";
static const char local_mcu_Pin0__s__7[] = ".isInput\n";
static const char local_mcu_Pin0__s__11[] = "Pin";
static const char local_mcu_Pin0__s__4[] = ".toggle\n";
static const char local_mcu_Pin0__s__8[] = ".makeOutput\n";
static const char local_mcu_Pin0__s__1[] = "\n";

/*
 * ======== function definitions ========
 */

void local_mcu_Pin0_set__E() {
    (local_mcu_Pin0_state__V) = true;
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin0_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin0__s__2);
}

void local_mcu_Pin0_makeOutput__E() {
    (local_mcu_Pin0_input__V) = false;
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin0_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin0__s__8);
}

bool local_mcu_Pin0_isInput__E() {
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin0_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin0__s__7);
    return(  (local_mcu_Pin0_input__V) == true  );
}

bool local_mcu_Pin0_isOutput__E() {
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin0_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin0__s__9);
    return(  (local_mcu_Pin0_input__V) == false  );
}

bool local_mcu_Pin0_get__E() {
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin0_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin0__s__5);
    return(  (local_mcu_Pin0_state__V)  );
}

void local_mcu_Pin0_clear__E() {
    (local_mcu_Pin0_state__V) = false;
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin0_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin0__s__3);
}

void local_mcu_Pin0_toggle__E() {
    (local_mcu_Pin0_state__V) = !(local_mcu_Pin0_state__V);
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin0_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin0__s__4);
}

void local_mcu_Pin0_targetInit__I() {
    (local_mcu_Pin0_name__V) = (local_mcu_Pin0_pname);
    (local_mcu_Pin0_num__V) = (local_mcu_Pin0_pnum);
    (local_mcu_Pin0_input__V) = false;
    (local_mcu_Pin0_state__V) = false;
    test90_Test_03_pollen__printStr((string)local_mcu_Pin0__s__0);
	test90_Test_03_pollen__printStr((string)(local_mcu_Pin0_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin0__s__1);
}

void local_mcu_Pin0_makeInput__E() {
    (local_mcu_Pin0_input__V) = true;
    test90_Test_03_pollen__printStr((string)(local_mcu_Pin0_pollen__unitname__V));
	test90_Test_03_pollen__printStr((string)local_mcu_Pin0__s__6);
}

